File: IR_VRML/IR/instantreality.org-tutorials/instantreality.org-tutorials/external-authoring-interface-javanet/EAIFramework.java
class EAIFramework
{
	public static void main(String[] args)
	{
		vrml.eai.Browser browser = null;
		try
		{
			java.net.InetAddress address = java.net.InetAddress.getByName("localhost");
			browser = vrml.eai.BrowserFactory.getBrowser(address, 4848);

			System.out.println("Browser.Name = \"" + browser.getName() + '"');
			System.out.println("Browser.Version = \"" + browser.getVersion() + '"');
			System.out.println("Browser.CurrentSpeed = " + browser.getCurrentSpeed());
			System.out.println("Browser.CurrentFrameRate = " + browser.getCurrentFrameRate());
			System.out.println("Browser.WorldURL = \"" + browser.getWorldURL() + '"');
		}
		catch (Throwable all)
		{
			all.printStackTrace();
		}
		finally
		{
			if (browser != null)
				browser.dispose();
		}
	}
}

File: IR_VRML/IR/instantreality.org-tutorials/instantreality.org-tutorials/external-authoring-interface-javanet/EAIFramework.cs
public class EAIFramework
{
    public static void Main(string[] args)
    {
        Vrml.EAI.Browser browser = null;
        try
        {
            System.Net.IPAddress address = System.Net.Dns.GetHostAddresses("localhost")[0];
            browser = Vrml.EAI.BrowserFactory.GetBrowser(address, 4848);

            System.Console.WriteLine("Browser.Name = \"" + browser.Name + '"');
            System.Console.WriteLine("Browser.Version = \"" + browser.Version + '"');
            System.Console.WriteLine("Browser.CurrentSpeed = " + browser.CurrentSpeed);
            System.Console.WriteLine("Browser.CurrentFrameRate = " + browser.CurrentFrameRate);
            System.Console.WriteLine("Browser.WorldURL = \"" + browser.WorldURL + '"');
        }
        catch (System.Exception ex)
        {
            System.Console.WriteLine(ex);
        }
        finally
        {
            if (browser != null)
                browser.Dispose();
        }
    }
}
